python/ray/tune/schedulers/pbt.py

```diff
@@ -393,3 +393,190 @@ def last_scores(self, trials):
+            ``~/ray_results``
+        trial_id (str): ID of the trial that should be replayed.
+
+    Example:
+
+    .. code-block:: python
+
+        # Replaying a result from ray.tune.examples.pbt_convnet_example
+        from ray import tune
+
+        from ray.tune.examples.pbt_convnet_example import PytorchTrainable
+        from ray.tune.schedulers import PopulationBasedTrainingReplay
+
+        replay = PopulationBasedTrainingReplay(
+            "~/ray_results/pbt_test/", "XXXXX_00001")
+
+        tune.run(
+            PytorchTrainable,
+            scheduler=replay,
+            stop={"training_iteration": 100})
+
+
+    """
+
+    def __init__(self, experiment_dir, trial_id):
+        experiment_dir = os.path.expanduser(experiment_dir)
+        if not os.path.exists(experiment_dir):
+            raise ValueError(
+                "Experiment directory not found: {}".format(experiment_dir))
+
+        self.experiment_dir = experiment_dir
+        self.trial_id = trial_id
+
+        # Find and read pbt policy file
+        initial_config, self._policy = self._load_policy(
+            self.experiment_dir, self.trial_id)
+        if not self._policy:
+            logger.warning(
+                "Empty PBT policy found for trial id {}. This means "
+                "the replay will only run with the initial config. "
+                "You might have specified a trial that did not exploit "
+                "other trials. Did you specify the correct trial?".format(
+                    self.trial_id))
+
+        self.experiment_tag = "replay_{}".format(self.trial_id)
+        self.config = initial_config
+        self.current_config = self.config
+
+        self._trial = None
+        self._current_step = 0
+        self._num_perturbations = 0
+
+        self._policy_iter = iter(self._policy)
+        self._next_policy = next(self._policy_iter, None)
+
+    def _load_policy(self, experiment_dir, trial_id):
+        raw_policy = []
+        pbt_policy_file = os.path.join(experiment_dir,
+                                       "pbt_policy_{}.txt".format(trial_id))
+        if not os.path.isfile(pbt_policy_file):
+            logger.warning(
+                "No PBT policy file found for trial id {}. This means "
+                "the replay will only run with the initial config. "
+                "You might have specified a trial that did not exploit "
+                "other trials. Did you specify the correct trial?".format(
+                    self.trial_id))
+            return {}, []
+        else:
+            with open(pbt_policy_file, "rt") as fp:
+                for row in fp.readlines():
+                    try:
+                        parsed_row = json.loads(row)
+                    except json.JSONDecodeError:
+                        raise ValueError(
+                            "Could not read PBT policy file: {}.".format(
+                                pbt_policy_file))
```

Comment:
this will end up raising multiple exceptions - try `from None` https://stackoverflow.com/a/25849414/2213289